exercises/complex-numbers/canonical-data.json

```diff
@@ -0,0 +1,251 @@
+        },
+        {
+          "description": "Absolute value of a purely imaginary number with negative imaginary part",
+          "property": "abs",
+          "input": [0, -5],
+          "expected": 5
+        },
+        {
+          "description": "Absolute value of a number with real and imaginary part",
+          "property": "abs",
+          "input": [3, 4],
+          "expected": 5
+        }
+      ]
+    },
+    {
+      "description": "Complex conjugate",
+      "cases": [
+        {
+          "description": "Conjugate a purely real number",
+          "property": "conjugate",
+          "input": [5, 0],
+          "expected": [5, 0]
+        },
+        {
+          "description": "Conjugate a purely imaginary number",
+          "property": "conjugate",
+          "input": [0, 5],
+          "expected": [0, -5]
+        },
+        {
+          "description": "Conjugate a number with real and imaginary part",
+          "property": "conjugate",
+          "input": [1, 1],
+          "expected": [1, -1]
+        }
+      ]
+    },
+    {
+      "description": "Real part",
+      "cases": [
+        {
+          "description": "Real part of a purely real number",
+          "property": "real",
+          "input": [1, 0],
+          "expected": 1
+        },
+        {
+          "description": "Real part of a purely imaginary number",
+          "property": "real",
+          "input": [0, 1],
+          "expected": 0
+        },
+        {
+          "description": "Real part of a number with real and imaginary part",
+          "property": "real",
+          "input": [1, 2],
+          "expected": 1
+        }
+      ]
+    },
+    {
+      "description": "Imaginary part",
+      "cases": [
+        {
+          "description": "Imaginary part of a purely real number",
+          "property": "imaginary",
+          "input": [1, 0],
+          "expected": 0
+        },
+        {
+          "description": "Imaginary part of a purely imaginary number",
+          "property": "imaginary",
+          "input": [0, 1],
+          "expected": 1
+        },
+        {
+          "description": "Imaginary part of a number with real and imaginary part",
+          "property": "imaginary",
+          "input": [1, 2],
+          "expected": 2
+        }
+      ]
+    },
+    {
+      "comments": [
+        " Defining the exponential function can be optional.       ",
+        " If the language used does not have sine and cosine       ",
+        " functions in the standard library, this will be          ",
+        " significantly more difficult than the rest of the exer-  ",
+        " cise and should probably not be part of the problem.     ",
+        " The recommended implementation uses Euler's formula      ",
+        " exp(ix) = cos(x) + i * sin(x). This is not an ideal sol- ",
+        " ution but works for the purpose of teaching complex      ",
+        " numbers.                                                 "
+      ],
```

Comment:
This is quite a limited test - perhaps add tests for exponentiation on real numbers, i^2, etc.